public/app/features/commandPalette/CommandPalette.tsx

```diff
@@ -65,7 +58,18 @@ export const CommandPalette = () => {
   useEffect(() => {
     if (showing) {
       reportInteraction('commandPalette_opened');
+
+      keybindingSrv.bindGlobal('esc', () => {
+        query.setVisualState(VisualState.animatingOut);
+      });
     }
+
+    return () => {
+      keybindingSrv.bindGlobal('esc', () => {
+        keybindingSrv.globalEsc();
+      });
```

Comment:
this feels messy, in that external consumers need to know to do this. hm.. 

Wonder how we can keep this internal to keybindingSrv somehow

Reply:
Could `.bindGlobal` return a function that when called would revert the state to the previous one?